packages/edit-post/src/style.scss

```diff
@@ -104,13 +104,56 @@ body.block-editor-page {
 .wp-block {
 	max-width: $content-width;
 
-	&[data-align="wide"],
-	&.alignwide {
+	&[data-align="wide"] {
 		max-width: 1100px;
 	}
 
-	&[data-align="full"],
-	&.alignfull {
+	&[data-align="full"] {
 		max-width: none;
 	}
 }
+
+.wp-align-wrapper {
+	max-width: $content-width;
+	margin-left: auto;
+	margin-right: auto;
+
+	> .wp-block,
+	&.wp-align-full {
+		max-width: none;
+	}
+
+	&.wp-align-full {
+		margin-left: -$block-padding;
+		margin-right: -$block-padding;
+
+		@include break-small() {
+			margin-left: -$block-padding - $block-padding - $block-side-ui-width - $border-width - $border-width;
+			margin-right: -$block-padding - $block-padding - $block-side-ui-width - $border-width - $border-width;
+		}
+	}
+
+	&.wp-align-wide {
+		max-width: 1100px;
+	}
+
+	&.wp-align-left > * {
+		float: left;
+		margin-right: 2em;
+	}
+
+	&.wp-align-right > * {
+		float: right;
+		margin-left: 2em;
+	}
+
+	&.wp-align-left > *,
+	&.wp-align-right > * {
+		// Without z-index, won't be clickable as "above" adjacent content.
+		z-index: z-index(".block-editor-block-list__block {core/image aligned left or right}");
+	}
+
+	&.wp-align-center {
+		// Same as no alignment?
```

Comment:
this needs display: flex and justify-content center for me.

Reply:
you should be able to see the difference between "center" and "no" when you resize an image smaller than the $content-width

Reply:
display: flex has the unfortunate side effect of not collapsing margins. :) I'll figure it out though.